### PR TITLE
Making popsplits more vanilla.

### PR DESCRIPTION
**Pull request by AlexHGaming**

### CODE REVIEW

**Comment by MastaCoder** [comment]
Anybody confirm? (I don't play Agar, can't tell what changed xD)

**Comment by Jhair2015** [comment]
@MastaCoder I will do it

**Comment by Agarian** [comment]
Is this after that "unshift" line of code was added to PlayerCell.js (or was it added at all yet)?

**Comment by AlexHGaming** [comment]
@Agarian Already added.

**Comment by AlexHGaming** [comment]
@Agarian Long distance popsplits improved, higher chance of having popsplits having 3 pieces.
Megabyte sent me the code on Discord, so I edited it and made this.

**Comment by Agarian** [comment]
Good

File: src/entity/Virus.js [hunk 1]
```diff
@@ -45,12 +44,12 @@ Virus.prototype.onEaten = function (c) {
...
-        var exp = Math.random() * (5 - 3.33) + 3.33;
-        while (threshold / exp > 24) {
-            threshold /= exp;
-            exp = 2;
-            big.push(threshold >> 0);
-        }
+        var exp = Math.random() * (4 - 3.33) + 3.33;
+while (threshold / exp > 24) {
+    threshold /= exp;
+    exp = Math.random() * (3 - 3) + 2;
```
Comment (Tyler3D) [inline src/entity/Virus.js]: Wouldn't this always generate 3?

Comment (MastaCoder) [inline src/entity/Virus.js]: (3 - 3) Well that part already doesn't make sense lmao.

**Comment by DeadLord1337** [comment]
Finally, you heard my message / wish !!! I hope the destruction from viruses will be better and similar to the original!
But I myself guess that these values) and very similar to the original

>     else if (c._size > 216) {
>         // virus explosion multipliers
>         var exp = Math.random() * (4 - 3.4) + 3.3;
>         while (threshold / exp > 24) {
>             threshold /= exp;
>             exp = 2;
>             big.push(threshold >> 0);

**Comment by Agarian** [comment]
Will someone merge this?

**Comment by FantasyIsBae** [comment]
@MastaCoder

**Comment by MastaCoder** [comment]
The merge doesn't make sense. 

1. Some parts of the code don't make sense, as stated in the comment above a bit back.
2. Formatting is off in lots of the code.

Until these are fixed I'm not merging this. (Also nobody really confirmed if it works and what it does lol)

**Comment by Agarian** [comment]
So, the `unshift` code that fixed doublesplits apparently messes with popsplits to the point where it's near impossible for one to end in 3 cells. That makes popsplits not only ugly, but unrealistic as well. This increases the chance for popsplits to end in 3 cells (not by much though), but it does. @MastaCoder 

I am also curious if @Luka967 knows about this issue, and/or knows how to fix it.

**Comment by MastaCoder** [comment]
Okay:

1. https://github.com/Megabyte918/MultiOgar-Edited/pull/723/commits/073192e57485cf266c8f149710ba8667904084be#diff-7742a26d82bd6315563b1d297a59169aL37 - Why delete that spacing, tbh I find it better if it's code, then a newline, and then a comment. Squishing comments together isn't really good formatting. 👍 (No biggy)
2. https://github.com/Megabyte918/MultiOgar-Edited/pull/723/commits/073192e57485cf266c8f149710ba8667904084be#diff-7742a26d82bd6315563b1d297a59169aR48 - Format
3. https://github.com/Megabyte918/MultiOgar-Edited/pull/723/commits/073192e57485cf266c8f149710ba8667904084be#diff-7742a26d82bd6315563b1d297a59169aR50 3 - 3 is 0, why do the calculation?
4. https://github.com/Megabyte918/MultiOgar-Edited/pull/723/commits/073192e57485cf266c8f149710ba8667904084be#diff-7742a26d82bd6315563b1d297a59169aR62 - I don't think this var is required if the code already works fine.

**Comment by Agarian** [comment]
@AlexHGaming have you considered using @DeadLord1337's values?
```js
else if (c._size > 216) {
    // virus explosion multipliers
    var exp = Math.random() * (4 - 3.4) + 3.3;
    while (threshold / exp > 24) {
        threshold /= exp;
        exp = 2;
        big.push(threshold >> 0);
```

**Comment by AlexHGaming** [comment]
@MastaCoder 

> 073192e#diff-7742a26d82bd6315563b1d297a59169aR50 3 - 3 is 0, why do the calculation?

This means that always 3 big pieces pop out. If I did 4 - 2, it would be random between 2 big pieces and 4, using Math.Random.

> I don't think this var is required if the code already works fine.

eh true, but it was used out of scope.

>  Why delete that spacing, tbh I find it better if it's code, then a newline, and then a comment. Squishing comments together isn't really good formatting. 👍 (No biggy)

I only just added what I preferred in my repo. You can change this tho. 👍

**Comment by AlexHGaming** [comment]
I also realised why popsplits hardly work. The unshifting in nodeplayer movement, does fix the doublesplitting however, but reduces the chance of hitting long distance and 3 piece popsplits alltogether. So it'd be better adding 'push' back and changing the doublesplits so it's biggest piece at front, while alongside with popsplits. lemme add a few things.

**Comment by MastaCoder** [comment]
About the math thing. Removing the Math.random() gives you:

(3 - 3) + 2;

3 - 3 is always going to be 0, why not just simplify to + 2?

**Comment by AlexHGaming** [comment]
@MastaCoder eh, true.
It was originally 4 - 2 as i set it, but I thought 3 - 3 is better. If they're the same value, then yeah you can remove math.random (my mistake) but if it's not the same, keep math.random.

**Comment by MastaCoder** [comment]
Okay I will wait until all issues are resolved.

> I also realised why popsplits hardly work. The unshifting in nodeplayer movement, does fix the doublesplitting however, but reduces the chance of hitting long distance and 3 piece popsplits alltogether. So it'd be better adding 'push' back and changing the doublesplits so it's biggest piece at front, while alongside with popsplits. lemme add a few things.

**Comment by AlexHGaming** [comment]
@MastaCoder They're in the other pull request. Don't merge the revertion, just merge the changes.

**Comment by AlexHGaming** [comment]
yay megerbites back

**Comment by Megabyte918** [comment]
@AlexHGaming 
> I also realised why popsplits hardly work. The unshifting in nodeplayer movement, does fix the doublesplitting however, but reduces the chance of hitting long distance and 3 piece popsplits alltogether. So it'd be better adding 'push' back and changing the doublesplits so it's biggest piece at front, while alongside with popsplits. lemme add a few things.

It does indeed reduce the chance of 3-piece popsplits working, however long distance seems to work  better from what I've seen (judging by the explosion multipliers on the latest version)

Reverting the commit that fixed doublesplit cell switching order would fix it, however there are many other solutions that could fix popsplits while also keeping that commit intact. Besides, cell switching order is a MUCH more annoying issue than popsplits imo. I recommend just keeping the fix there and finding a workaround for popsplits later. Also you can delete commits by using git or just reverting the changes you made would be fine in #717

> 073192e#diff-7742a26d82bd6315563b1d297a59169aR50 3 - 3 is 0, why do the calculation?
This means that always 3 big pieces pop out. If I did 4 - 2, it would be random between 2 big pieces and 4, using Math.Random.

Incorrect. It would be best to keep the second multiplier at 2 since the line you are using always generates 2 anyway.

Also one more issue, the virus explosion multipliers you guys are using makes virus explosions no longer like vanilla. Both you and @DeadLord1337's code makes too many cells popout when you are small and big which results in weird looking explosions not similar to those on vanilla servers. It's best to keep the ratio between what looks like vanilla and what is best for popsplits around the same. That's what I've been trying to do all along.

**Comment by AlexHGaming** [comment]
@Megabyte918 ok i was wrong lol, i thought something else

**Comment by AlexHGaming** [comment]
Re-opening this when everything is fixed.

**Pull request closed without merging**